ethereum/spec/src/main/java/tech/pegasys/teku/spec/schemas/registry/BaseSchemaProvider.java

```diff
@@ -90,100 +96,104 @@ public Set<SpecMilestone> getSupportedMilestones() {
+   * Creates a builder for a schema provider.<br>
    * Example usage:
    *
    * <pre>{@code
-   * constantProviderBuilder(EXAMPLE_SCHEMA)
-   *    .withCreator(ALTAIR, (registry, config) -> new ExampleSchemaAltair())
-   *    .withCreator(ELECTRA, (registry, config) -> new ExampleSchemaElectra())
+   * variableProviderBuilder(EXAMPLE_SCHEMA)
+   *    .withCreator(ALTAIR, (registry, config) -> ExampleSchema1.create(registry, config))
+   *    .withCreator(ELECTRA, (registry, config) -> ExampleSchema2.create(registry, config))
    *    .build();
    *
    * }</pre>
    *
    * this will create a schema provider that will generate: <br>
-   * - only one ExampleSchemaAltair instance which will be reused from ALTAIR to BELLATRIX <br>
-   * - only one ExampleSchemaElectra instance which will be reused from ELECTRA to last known
-   * milestone <br>
-   */
-  static <T> Builder<T> constantProviderBuilder(final SchemaId<T> schemaId) {
-    return new Builder<>(schemaId, true);
-  }
-
-  /**
-   * Creates a builder for a variable schema provider.<br>
-   * This can be used when schema changes across multiple milestones (i.e. depends on changing
-   * schemas) <br>
-   * Example usage:
+   * - a new ExampleSchema1 for each milestone from ALTAIR to CAPELLA <br>
+   * - a new ExampleSchema2 for each milestone from ELECTRA to last known milestone <br>
+   *
+   * <p>By default, the schema provider will check for schema equality when a creator is used
+   * multiple times. In the previous example, if ExampleSchema1.create generates schemas that are
+   * <b>equals</b> in both ALTAIR and BELLATRIX context, the ALTAIR instance will be used for
+   * BELLATRIX too.<br>
+   * Since the equality check does not consider names, semantically equivalent schemas with
+   * different fields or container names will be considered equal.<br>
+   *
+   * <p>If the equality check is relevant, this behavior can be avoided in two ways:<br>
+   * - specifying a new creator like: <br>
    *
    * <pre>{@code
    * variableProviderBuilder(EXAMPLE_SCHEMA)
-   *    .withCreator(ALTAIR, (registry, config) -> new ExampleSchema1(registry, config))
-   *    .withCreator(ELECTRA, (registry, config) -> new ExampleSchema2(registry, config))
+   *    .withCreator(ALTAIR, (registry, config) -> ExampleSchema1.create(registry, config))
+   *    .withCreator(BELLATRIX, (registry, config) -> ExampleSchema1.create(registry, config))
+   *    .withCreator(ELECTRA, (registry, config) -> ExampleSchema2.create(registry, config))
    *    .build();
    *
    * }</pre>
    *
-   * this will create a schema provider that will generate: <br>
-   * - a new instance of ExampleSchema1 for each milestone from ALTAIR to ELECTRA <br>
-   * - a new instance of ExampleSchema2 for each milestone from ELECTRA to last known milestone <br>
+   * - disabling the schema equality check by calling {@link Builder#disableSchemaEqualityCheck()}
```

Comment:
it's not clear for me what disableSchemaEqualityCheck does. Maybe we renaming/changing the doc towards outcome of its usage rather than logic of builder change could make sense?